src/olympia/activity/utils.py

```diff
@@ -24,6 +24,7 @@
 REPLY_TO_PREFIX = 'reviewreply+'
 # Group for users that want a copy of all Activity Emails.
 ACTIVITY_MAIL_GROUP = 'Activity Mail CC'
+NOTIFICATIONS_FROM_EMAIL = 'notifications@addons.mozilla.org'
```

Comment:
Don't hard-code the site address.  You should be concatenating `notifications@` to `settings.INBOUND_EMAIL_DOMAIN`